compiler/rustc_session/src/filesearch.rs

```diff
@@ -182,7 +182,15 @@ pub fn get_or_default_sysroot() -> Result<PathBuf, String> {
         if dir.ends_with(crate::config::host_triple()) {
             dir.parent() // chop off `$target`
                 .and_then(|p| p.parent()) // chop off `rustlib`
-                .and_then(|p| p.parent()) // chop off `lib`
+                .and_then(|p| {
+                    match p.parent() {
+                        Some(p) => match p.file_name() {
+                            Some(f) if f != "lib" => p.parent(), // chop off `lib` and the $arch folder
+                            _ => Some(p),                        // chop off `lib` only
```

Comment:
```suggestion
                    match p.parent() { // chop off `lib` (this could be also $arch dir)
                        Some(p) => match p.file_name() {
                            Some(f) if f != "lib" => p.parent(), // first chop went for $arch, so chap off again for `lib`
                            _ => Some(p),
```

Reply:
Thank you for the suggestion!